File: src/org/jetbrains/jet/j2k/ast/CaseContainer.java
package org.jetbrains.jet.j2k.ast;

import org.jetbrains.annotations.NotNull;

import java.util.LinkedList;
import java.util.List;

/**
 * @author ignatov
 */
public class CaseContainer extends Statement {
  private Statement myCaseStatement;
  private Block myBlock;

  public CaseContainer(final Statement caseStatement, final List<Statement> statements) {
    myCaseStatement = caseStatement;
    List<Statement> newStatements = new LinkedList<Statement>();
    for (Statement s : statements)
      if (s.getKind() != Kind.BREAK && s.getKind() != Kind.CONTINUE )
        newStatements.add(s);
    myBlock = new Block(newStatements);
  }

  @NotNull
  @Override
  public String toKotlin() {
    return myCaseStatement.toKotlin() + SPACE + "=>" + SPACE + myBlock.toKotlin();
  }
}

File: src/org/jetbrains/jet/j2k/ast/CaseContainer.kt
package org.jetbrains.jet.j2k.ast

import org.jetbrains.jet.j2k.util.AstUtil
import java.util.LinkedList
import java.util.List

public open class CaseContainer(val caseStatement: List<Statement>, statements: List<Statement>): Statement() {
    private val myBlock: Block

    {
        val newStatements: List<Statement> = statements.filterNot { it is BreakStatement || it is ContinueStatement }
        myBlock = Block(newStatements, false)
    }

    public override fun toKotlin(): String {
        return AstUtil.joinNodes(caseStatement, ", ") + " -> " + myBlock.toKotlin()
    }
}
